### ISSUE TITLE
How Detectron2 does binary_cross_entropy_with_logits ?

### ISSUE THREAD

**Issue by Robert-BoMiao** [facebookresearch/detectron2#1121]
Hi,

Code location:
detectron2/modeling/roi_heads/mask_head.py

1. I noticed from the codes that mask head put pred_mask_logits and gt_masks into F.binary_cross_entropy_with_logits to calculate loss. However, the pred_mask_logits need a sigmoid layer to finish the prediction. Does F.binary_cross_entropy_with_logits inherently contain a sigmoid layer for the calculation of loss? 

mask_loss = F.binary_cross_entropy_with_logits(pred_mask_logits, gt_masks, reduction="mean")

2. I found that when calculating mask_incorrect for accuracy, the pred_mask_logits only use a threshold 0.0 to build the bool mask. However, from my knowledge, I think pred_mask_logits should through a sigmoid layer and use a threshold 0.5 to build the bool mask. 

mask_incorrect = (pred_mask_logits > 0.0) != gt_masks_bool

Thanks.

**Comment by ppwwyyxx** [comment]
1. it's a pytorch question unrelated to this project
2. sigmoid(0.0)=0.5
